src/java.base/share/classes/java/lang/Module.java

```diff
@@ -1590,8 +1590,8 @@ private Class<?> loadModuleInfoClass() {
      */
     private Class<?> loadModuleInfoClass(InputStream in) throws IOException {
         final String MODULE_INFO = "module-info";
-        byte[] bytes = Classfile.parse(in.readAllBytes(),
-                Classfile.Option.constantPoolSharing(false)).transform((clb, cle) -> {
+        var cc = Classfile.of(Classfile.ConstantPoolSharingOption.DO_NOT_SHARE_CONSTANT_POOL);
```

Comment:
This `cc` can be stored in a static final field instead.

Reply:
Yes, we can avoid repeated construction of default class hierarchy resolver here.

Reply:
In this situation the ClassHierarchyResolver is never used and have no caching function.
Adding static field with initializer does not make much sense here.